FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeProcessingJobSpliter.cs

```diff
@@ -0,0 +1,291 @@
+                    value = anchorList[item.Key];
+                }
+
+                if (value - baseSize < LowBoundOfProcessingJobResourceCount)
+                {
+                    last = item.Key;
+                    continue;
+                }
+
+                if (value - baseSize <= HighBoundOfProcessingJobResourceCount && value - baseSize >= LowBoundOfProcessingJobResourceCount)
+                {
+                    return item.Key;
+                }
+
+                return await BisectAnchor(resourceType, last == null ? DateTimeOffset.MinValue : (DateTimeOffset)last, item.Key, anchorList, baseSize, cancellationToken);
+            }
+
+            return last;
+        }
+
+        private async Task<DateTimeOffset> BisectAnchor(string resourceType, DateTimeOffset start, DateTimeOffset end, Dictionary<DateTimeOffset, int> anchorList, int baseSize, CancellationToken cancellationToken)
+        {
+            while ((end - start).TotalMilliseconds > 1)
+            {
+                DateTimeOffset mid = start.Add((end - start) / 2);
+                var resourceCount = await GetResourceCountAsync(resourceType, start, mid, cancellationToken);
+                resourceCount = resourceCount == int.MaxValue ? int.MaxValue : resourceCount + anchorList[start];
+                anchorList[mid] = resourceCount;
+                if (resourceCount - baseSize > HighBoundOfProcessingJobResourceCount)
+                {
+                    end = mid;
+                }
+                else if (resourceCount - baseSize < LowBoundOfProcessingJobResourceCount)
+                {
+                    start = mid;
+                }
+                else
+                {
+                    return mid;
+                }
+            }
+
+            return end;
+        }
+
+        private async Task<int> GetResourceCountAsync(string resourceType, DateTimeOffset? start, DateTimeOffset end, CancellationToken cancellationToken)
+        {
+            List<KeyValuePair<string, string>> parameters = new List<KeyValuePair<string, string>>
+            {
+                new KeyValuePair<string, string>(FhirApiConstants.LastUpdatedKey, $"lt{end.ToInstantString()}"),
+                new KeyValuePair<string, string>(FhirApiConstants.SummaryKey, "count"),
+                new KeyValuePair<string, string>(FhirApiConstants.TypeKey, resourceType),
+            };
+
+            if (start != null)
+            {
+                parameters.Add(new KeyValuePair<string, string>(FhirApiConstants.LastUpdatedKey, $"ge{((DateTimeOffset)start).ToInstantString()}"));
+            }
+
+            var searchOptions = new BaseSearchOptions(null, parameters);
+            string fhirBundleResult;
+            try
+            {
+                fhirBundleResult = await _dataClient.SearchAsync(searchOptions, cancellationToken);
+            }
+            catch
```

Comment:
do we need to catch the specific exceptions for too much resources?

Reply:
Added exception handling here, thanks.